src/eng_back.c

```diff
@@ -241,7 +241,13 @@ ENGINE_CTX *ctx_new()
 	ctx->rwlock = CRYPTO_get_dynlock_create_callback() ?
 		CRYPTO_get_new_dynlockid() : 0;
 #endif
-
+/*
+ * OpenSSL Version prior to 1.1.0 are known to work
+ * with random.
+ */
+#if OPENSSL_VERSION_NUMBER < 0x101010000
```

Comment:
This looks reasonable

Reply:
I'm not sure if I have this version number right. I checked out 1.1.0 and used that version number...so I think I do.

I wonder if the issues we're seeing has to do with:
https://github.com/openssl/openssl/blob/master/crypto/rand/rand_lib.c#L817
Ihaven't looked at it in detail though.